### PR TITLE
add linked hash set to builtin

### PR DESCRIPTION
**Pull request by yj-qin**
related issue: #1070

### CODE REVIEW

File: hashmap/hashmap_test.mbt [hunk 1]
```diff
@@ -143,11 +143,11 @@ test "iter" {
 }
 
 test "to_array" {
-  let map = of([(1, "one"), (2, "two"), (3, "three")])
+  let map = @hashmap.of([(1, "one"), (2, "two"), (3, "three")])
   inspect!(
     map.to_array(),
     content=
-      #|[(1, "one"), (2, "two"), (3, "three")]
+      #|[(2, "two"), (1, "one"), (3, "three")]
```
Comment (qazxcdswe123) [inline hashmap/hashmap_test.mbt:150]: Why is this test changed though

File: builtin/linked_hash_set.mbt [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+/// Create a hash set from array.
+pub fn Set::from_array[K : Hash + Eq](arr : Array[K]) -> Set[K] {
+  let m = Set::new(capacity=arr.length())
+  arr.each(fn(e) { m.insert(e) })
+  m
+}
+
+/// Insert a key into the hash set.
+/// @alert unsafe "Panic if the hash set is full."
```
Comment (qazxcdswe123) [inline builtin/linked_hash_set.mbt:72]: I couldn't find the panic here

File: hashmap/hashmap_test.mbt (hunk 1 above)
Comment (yj-qin) [reply hashmap/hashmap_test.mbt:150]: This test actually use the `of` method of builtin linked hash map, it should use the `of` method in the `hashmap` pacakage. So I change it.

**Comment by Yoorkin** [comment]
Note that the names of Set operation are different in `@sorted_set` `@immut/sorted_set` `@hashset` and this linked hash set, we need to keep the API design consistent.

**Comment by Yoorkin** [comment]
Here is an incomplete list of different APIs @qazxcdswe123 

```
// in this PR
  insert[K : Hash + Eq](Self[K], K) -> Unit
  difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  intersection[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  symmetric_difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  union[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
```

```
// @sorted_set
  add[V : Compare + Eq](Self[V], V) -> Unit
  diff[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
  disjoint[V : Compare + Eq](Self[V], Self[V]) -> Bool
  intersect[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
  subset[V : Compare + Eq](Self[V], Self[V]) -> Bool  // is_subset?
  union[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
```

```
// @immut/sorted_set
  add[A : Compare + Eq](Self[A], A) -> Self[A]
  contains[A : Compare + Eq](Self[A], A) -> Bool
  diff[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
  disjoint[A : Compare + Eq](Self[A], Self[A]) -> Bool
  inter[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
  subset[A : Compare + Eq](Self[A], Self[A]) -> Bool
  union[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
```

```
// @hashset
  insert[K : Hash + Eq](Self[K], K) -> Unit
  difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  intersection[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  symmetric_difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
  union[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
```

**Comment by qazxcdswe123** [comment]
> Here is an incomplete list of different APIs @qazxcdswe123
> 
> ```
> // in this PR
>   insert[K : Hash + Eq](Self[K], K) -> Unit
>   difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   intersection[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   symmetric_difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   union[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
> ```
> 
> ```
> // @sorted_set
>   add[V : Compare + Eq](Self[V], V) -> Unit
>   diff[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
>   disjoint[V : Compare + Eq](Self[V], Self[V]) -> Bool
>   intersect[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
>   subset[V : Compare + Eq](Self[V], Self[V]) -> Bool  // is_subset?
>   union[V : Compare + Eq](Self[V], Self[V]) -> Self[V]
> ```
> 
> ```
> // @immut/sorted_set
>   add[A : Compare + Eq](Self[A], A) -> Self[A]
>   contains[A : Compare + Eq](Self[A], A) -> Bool
>   diff[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
>   disjoint[A : Compare + Eq](Self[A], Self[A]) -> Bool
>   inter[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
>   subset[A : Compare + Eq](Self[A], Self[A]) -> Bool
>   union[A : Compare + Eq](Self[A], Self[A]) -> Self[A]
> ```
> 
> ```
> // @hashset
>   insert[K : Hash + Eq](Self[K], K) -> Unit
>   difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   intersection[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   symmetric_difference[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
>   union[K : Hash + Eq](Self[K], Self[K]) -> Self[K]
> ```

If we have `linked hashse`t maybe we can eliminate `sorted_set` and `hashset`?

**Comment by bobzhang** [comment]
> If we have linked hashset maybe we can eliminate sorted_set and hashset?

They serve different purposes, the one in builtin is the mostly used one

**Pull request merged**